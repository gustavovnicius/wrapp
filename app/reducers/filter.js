import {
  APPLY_FILTER,
} from 'actions/filter';
import {
  defaultFilter,
  enabledFilters,
} from 'plugins/filter';

const initialState = {
  enabled: enabledFilters(),
  selected: defaultFilter(),
};

export default function filter(state = initialState, action) {
  switch (action.type) {
    case APPLY_FILTER:
      return {
        ...state,
        selected: action.payload.filter,
      };
    default:
      return state;
  }
}
