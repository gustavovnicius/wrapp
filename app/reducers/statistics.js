import _ from 'lodash';
import {
  ADD_STATISTIC,
} from 'actions/statistics';

const initialState = {
  statistics: [],
};

export default function statistics(state = initialState, action) {
  switch (action.type) {
    case ADD_STATISTIC:
      return {
        ...state,
        statistics: _.concat(state.statistics, [action.payload])
      }
    default:
      return state;
  }
}
