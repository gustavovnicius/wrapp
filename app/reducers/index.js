import { combineReducers } from 'redux';
import filter from 'reducers/filter';
import statistics from 'reducers/statistics';

export default combineReducers({
  filter,
  statistics,
});
