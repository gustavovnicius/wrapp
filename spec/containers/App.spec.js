import React from 'react';
import {
  Provider,
} from 'react-redux';
import {
  createStore,
} from 'redux';
import {
  App,
  mapStateToProps,
} from 'containers/App';
import reducers from 'reducers';
import renderer from 'react-test-renderer';

describe('App', () => {
  const statisticsData = [
    {
      name: 'A name',
      balance: 1000,
    },
    {
      name: 'Another name',
      balance: 2000,
    }
  ];

  it('match the snapshot', () => {
    const store = createStore(reducers);
    const app = renderer.create(
      <Provider store={store}>
        <App statistics={statisticsData} addStatistic={() => {}} />
      </Provider>
    ).toJSON();

    expect(app).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    it('map statistics data to component props', () => {
      const state = {
        statistics: {
          data: []
        }
      };
      const expected = {
        statistics: []
      };

      expect(mapStateToProps(state)).toEqual(expected);
    })
  });
});
