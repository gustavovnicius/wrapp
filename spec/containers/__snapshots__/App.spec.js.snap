// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App match the snapshot 1`] = `
<div
  className="Container"
  style={
    Object {
      "boxSizing": "border-box",
      "margin": "auto",
      "maxWidth": 1024,
      "paddingLeft": 16,
      "paddingRight": 16,
    }
  }
>
  <div
    className="Flex"
    style={
      Object {
        "WebkitAlignItems": "center",
        "WebkitJustifyContent": "center",
        "alignItems": "center",
        "boxSizing": "border-box",
        "display": "flex",
        "justifyContent": "center",
        "msAlignItems": "center",
        "msJustifyContent": "center",
      }
    }
  >
    <h1>
      Statistics
    </h1>
  </div>
  <div
    className="Flex"
    style={
      Object {
        "WebkitAlignItems": "center",
        "WebkitFlexDirection": "column",
        "WebkitJustifyContent": "center",
        "alignItems": "center",
        "boxSizing": "border-box",
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "center",
        "msAlignItems": "center",
        "msFlexDirection": "column",
        "msJustifyContent": "center",
      }
    }
  >
    <div>
      <ul>
        <li>
          <p>
            Name: 
            A name
          </p>
          <p>
            Balance: 
            1000
          </p>
        </li>
        <li>
          <p>
            Name: 
            Another name
          </p>
          <p>
            Balance: 
            2000
          </p>
        </li>
      </ul>
    </div>
    <div
      className="Container"
      style={
        Object {
          "boxSizing": "border-box",
          "margin": "auto",
          "maxWidth": 1024,
          "paddingLeft": 16,
          "paddingRight": 16,
        }
      }
    >
      <div
        className="Input"
        style={
          Object {
            "boxSizing": "border-box",
            "color": null,
            "marginBottom": 16,
          }
        }
      >
        <label
          className="Label"
          htmlFor="name"
          style={
            Object {
              "boxSizing": "border-box",
              "fontSize": 14,
              "fontWeight": 600,
              "lineHeight": 1,
            }
          }
        >
          Name
        </label>
        <input
          name="name"
          onChange={[Function]}
          placeholder="Statistic name"
          style={
            Object {
              "backgroundColor": "rgba(255, 255, 255, .25)",
              "borderColor": "rgba(0, 0, 0, .25)",
              "borderRadius": 2,
              "borderStyle": "solid",
              "borderWidth": 1,
              "boxSizing": "border-box",
              "color": "inherit",
              "display": "block",
              "fontFamily": "inherit",
              "fontSize": "inherit",
              "height": 32,
              "margin": 0,
              "paddingLeft": 8,
              "paddingRight": 8,
              "width": "100%",
            }
          }
          type="text"
          value=""
        />
      </div>
      <div
        className="Input"
        style={
          Object {
            "boxSizing": "border-box",
            "color": null,
            "marginBottom": 16,
          }
        }
      >
        <label
          className="Label"
          htmlFor="balance"
          style={
            Object {
              "boxSizing": "border-box",
              "fontSize": 14,
              "fontWeight": 600,
              "lineHeight": 1,
            }
          }
        >
          Balance
        </label>
        <input
          name="balance"
          onChange={[Function]}
          placeholder="Statistic balance"
          style={
            Object {
              "backgroundColor": "rgba(255, 255, 255, .25)",
              "borderColor": "rgba(0, 0, 0, .25)",
              "borderRadius": 2,
              "borderStyle": "solid",
              "borderWidth": 1,
              "boxSizing": "border-box",
              "color": "inherit",
              "display": "block",
              "fontFamily": "inherit",
              "fontSize": "inherit",
              "height": 32,
              "margin": 0,
              "paddingLeft": 8,
              "paddingRight": 8,
              "width": "100%",
            }
          }
          type="number"
          value=""
        />
      </div>
      <button
        className="Button"
        href={undefined}
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#08e",
            "border": 0,
            "borderRadius": 2,
            "boxSizing": "border-box",
            "color": "#fff",
            "cursor": "pointer",
            "display": "inline-block",
            "fontFamily": "inherit",
            "fontSize": 14,
            "fontWeight": 600,
            "lineHeight": "16px",
            "margin": 0,
            "minHeight": 32,
            "paddingBottom": 8,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 8,
            "textDecoration": "none",
          }
        }
      >
        Add
      </button>
    </div>
  </div>
</div>
`;
